price_pusher/src/index.ts

```diff
@@ -79,102 +76,99 @@ const argv = yargs(hideBin(process.argv))
   })
   .parseSync();
 
-let pythContractAddr: string;
-
-if (CONTRACT_ADDR[argv.pythContract] !== undefined) {
-  pythContractAddr = CONTRACT_ADDR[argv.pythContract];
-} else {
-  pythContractAddr = argv.pythContract;
-}
-
 const priceConfigs = readPriceConfigFile(argv.priceConfigFile);
 
-async function injectiveRun() {
-  const connection = new PriceServiceConnection(argv.priceEndpoint, {
-    logger: console,
-  });
-
-  const pythPriceListener = new PythPriceListener(connection, priceConfigs);
-
-  const injectivePriceListener = new InjectivePriceListener(
-    pythContractAddr,
-    argv.endpoint,
-    priceConfigs,
-    { pollingFrequency: argv.pollingFrequency }
-  );
-
-  const injectivePricePusher = new InjectivePricePusher(
-    connection,
-    argv.pythContract,
-    argv.endpoint,
-    fs.readFileSync(argv.mnemonicFile, "utf-8").trim()
-  );
-
+// TODO: name ChainPricePusher -> IPricePusher in a clean up PR
+// TODO: update listeners to not depend on the whole priceConfig
+async function start({
+  sourcePriceListener,
+  targetPriceListener,
+  targetPricePusher,
+}: {
+  sourcePriceListener: IPriceListener;
+  targetPriceListener: IPriceListener;
+  targetPricePusher: ChainPricePusher;
+}) {
   const handler = new Controller(
     priceConfigs,
-    pythPriceListener,
-    injectivePriceListener,
-    injectivePricePusher,
+    sourcePriceListener,
+    targetPriceListener,
+    targetPricePusher,
     {
       cooldownDuration: argv.cooldownDuration,
     }
   );
 
-  await injectivePriceListener.start();
-  await pythPriceListener.start();
+  await sourcePriceListener.start();
+  await targetPriceListener.start();
```

Comment:
maybe handler.start() should start the inner listeners too? that may simplify the usage.

Reply:
yeah, it does